Elasticsearch Devservices (#23753)
quarkusio/quarkus: docs/src/main/asciidoc/elasticsearch.adoc

If you need a more advanced configuration, you can find the comprehensive list of supported configuration properties at the end of this guide.

[[dev-services]]
=== Dev Services (Configuration Free Databases)

> Quarkus supports a feature called Dev Services that allows you to create various datasources without any config.

Reviewer: ```suggestion
Quarkus supports a feature called Dev Services that allows you to start various containers without any config.
```

In the case of Elasticsearch this support extends to the default Elasticsearch connection.

> What that means practically, is that if you have not configured `quarkus.elasticsearch.hosts` Quarkus will automatically

Reviewer: ```suggestion
What that means practically is that, if you have not configured `quarkus.elasticsearch.hosts`, Quarkus will automatically
```

> start an Elasticsearch container when running tests or dev-mode, and automatically configure the connection.

Reviewer: ```suggestion
start an Elasticsearch container when running tests or dev mode, and automatically configure the connection.
```

> When running the production version of the application, the Elasticsearch connection need to be configured as normal,

Reviewer: ```suggestion
When running the production version of the application, the Elasticsearch connection needs to be configured as usual,
```

so if you want to include a production database config in your `application.properties` and continue to use Dev Services
we recommend that you use the `%prod.` profile to define your Elasticsearch settings.

> For more information you can read the xref:elasticsearch-dev-services.adoc[Dev Services for Elasticsearch].

Reviewer: ```suggestion
For more information you can read the xref:elasticsearch-dev-services.adoc[Dev Services for Elasticsearch guide].
```
